contrib/packaging/rpm/finch.spec

```diff
@@ -0,0 +1,266 @@
+
+# cosign
+%global cosign_release 2.4.0
+%global cosign_commit b5e7dc123a272080f4af4554054797296271e902
+%global cosign_package github.com/sigstore/cosign
+%global cosign_src cosign-%{cosign_commit}
+%global cosign_rpm_version %(r=%cosign_release; echo ${r%%%%-*})
+
+Name: runfinch-finch
+Version: %{finch_rpm_version}
+Release: 1%{?dist}%{?_trivial}%{?_buildid}
+# Upstream license specification: Apache-2.0
+License: ASL 2.0
+Summary: The Finch CLI is an open source client for container development tool
+
+%if %{defined build_latest}
+Source0: https://github.com/runfinch/finch/archive/refs/heads/%{latest_branch}.tar.gz
+%else
+%if %{defined build_local}
+Source0: finch-%{finch_release}.tar.gz
+%else
+Source0: https://%{finch_package}/archive/%{finch_commit}/%{finch_release}.tar.gz
+%endif
+%endif
+Source1: finch.service
+Source2: finch.yaml
+Source3: nerdctl.toml
+Source4: buildkitd.toml
+Source5: finch-buildkit.service
+Source6: finch-buildkit.socket
+Source7: finch-soci.service
+Source8: finch-soci.socket
+Source9: soci-snapshotter-grpc.toml
+
+# buildkit
+Source1000: https://%{buildkit_package}/archive/%{buildkit_commit}/%{buildkit_src}.tar.gz
+
+# soci-snapshotter
+Source2000: https://%{soci_package}/archive/%{soci_commit}/%{soci_src}.tar.gz
+
+# cosign
+Source3000: https://%{cosign_package}/archive/%{cosign_commit}/%{cosign_src}.tar.gz
+
+# Runtime requirements
+Requires: containerd nerdctl cni-plugins
+
+Provides: finch = %{finch_rpm_version}
+# License: ASL 2.0
+Provides: bundled(finch-daemon) = %{finch_daemon_rpm_version}
+# License: ASL 2.0
+Provides: bundled(buildctl) = %{buildkit_rpm_version}
+# License: ASL 2.0
+Provides: bundled(buildkitd) = %{buildkit_rpm_version}
+# License: ASL 2.0
+Provides: bundled(cosign) = %{cosign_rpm_version}
+
+# Compilation requirements
+BuildRequires: golang >= 1.22.3, git, make
+%if 0%{?amzn} > 2
+# this macro package doesn't exist on AL2
+BuildRequires: systemd-rpm-macros
+%endif
+
+%description
+Finch is an open source client for container development. Its simple
+installer provides a minimal native client along with an opinionated
+distribution of other open source components. Rather than creating
+even more options to reason about and choose from, Finch aims to help
+promote other projects by making it easy to install and use them,
+while offering a simple native client to tie it all together.
+
+%prep
+# %setup -D -T -c -b 0 -n finch-%{finch_commit}
+%setup -c
+%if %{defined build_latest}
+mv "finch-%{latest_branch}" "%{finch_src}"
+%endif
+
+# extract buildkit
+%setup -D -T -a 1000
+# extract soci
+%setup -D -T -a 2000
+# extract cosign
+%setup -D -T -a 3000
+
+%build
+
+# build finch
+pushd "%{finch_src}"
+VERSION=%{finch_release} GITCOMMIT=%{finch_commit} make finch
+popd
+
+# build buildkit
+pushd "%{buildkit_src}"
+go build -mod=vendor -buildmode=pie -ldflags '-w -X %{buildkit_package}/version.Version=%{buildkit_rpm_version}' -o _output/buildkitd %{buildkit_package}/cmd/buildkitd
```

Comment:
do we need to define a min go version?

Reply:
I think we will just keep it as low as possible, it will depend on when different dependencies upgrade